R/utils.R

```diff
@@ -16,6 +16,16 @@ parse_answer = function(api_answer, multiple = FALSE) {
 
   xml_source = content(api_answer, encoding = "ISO-8859-1")
 
+  apicontrol = xml_text(xml_find_all(xml_source, "//apicontrol"))
+
+  if (apicontrol == "invalid") {
+    stop("The query is invalid")
+  } else if (apicontrol == "invalidkey") {
+    stop("The provided API key is invalid. ",
```

Comment:
What about turning this into a simple `warning` and performing the request with the `ak` parameter stripped?

Reply:
The only way to control the key is to get this header back... So basically it would require one query to determine that the API key is invalid, then an additional one to get the results.
That's why I preferred to implement the `stop()` condition in `parse_answer()` instead of elsewhere?
Otherwise we could implement it in `check_key()` but that would be expansive if there are a lot of queries...

Reply:
Yes, you're right. That's an unfortunate behaviour but it should probably be changed server side instead of us trying to work around it. So let's keep your code.